ntoskrnl/cc/view.c

```diff
@@ -647,20 +648,31 @@ CcRosLookupVacb (
                            VACB_MAPPING_GRANULARITY,
                            FileOffset))
         {
-            CcRosVacbIncRefCount(current);
-            KeReleaseSpinLockFromDpcLevel(&SharedCacheMap->CacheMapLock);
-            KeReleaseQueuedSpinLock(LockQueueMasterLock, oldIrql);
-            return current;
+            Ret = current;
+            break;
         }
+
         if (current->FileOffset.QuadPart > FileOffset)
             break;
+
         current_entry = current_entry->Flink;
     }
 
     KeReleaseSpinLockFromDpcLevel(&SharedCacheMap->CacheMapLock);
+
+    if (Ret)
+    {
+        /* Move to the tail of the LRU list */
+        RemoveEntryList(&Ret->VacbLruListEntry);
+        InsertTailList(&VacbLruListHead, &Ret->VacbLruListEntry);
```

Comment:
Question, to be sure: is this wishable to have `CcRosLookupVacb()` also move the found VACB to the tail of the LRU list, even when tht function is not called as part of `CcRosGetVacb()` ?

Reply:
`CcRosLookupVacb` success means that the returned VACB is in use. It should be marked as most-recently used.